homeassistant/components/starlink/coordinator.py

```diff
@@ -0,0 +1,89 @@
+"""Contains the shared Coordinator for Starlink systems."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+from starlink_grpc import ChannelContext, GrpcError, get_status
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .dish_status import DishyStatus
+
+_LOGGER = logging.getLogger(__name__)
+
+DISHY_ID = "dishy_id"
+DISHY_HARDWARE = "dishy_hardware"
+
+
+class StarlinkUpdateCoordinator(DataUpdateCoordinator[DishyStatus]):
+    """Coordinates updates between all Starlink sensors defined in this file."""
+
+    def __init__(
+        self, hass: HomeAssistant, name: str, url: str, poll_rate: int
+    ) -> None:
+        """Initialize an UpdateCoordinator for a group of sensors."""
+        self.channel_context = ChannelContext(target=url)
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=name,
+            update_interval=timedelta(seconds=poll_rate),
+            request_refresh_debouncer=Debouncer(
+                hass=hass, logger=_LOGGER, cooldown=1, immediate=True
+            ),
+        )
+
+    def get_device(self) -> DeviceInfo:
+        """Get a DeviceInfo for Dishy."""
+        config_url = "http://" + self.channel_context.target.split(":")[0]
+        return DeviceInfo(
+            identifiers={
+                (DISHY_ID, self.data.id),
+                (DISHY_HARDWARE, self.data.hardware_version),
+            },
+            sw_version=self.data.software_version,
+            hw_version=self.data.hardware_version,
+            name=self.name,
+            configuration_url=config_url,
+            manufacturer="SpaceX",
+            model="Starlink",
+        )
```

Comment:
This could be part of a base entity instead (create an `entity.py` file and set the device info in that base entity. Extend all other platform entities from this case entity).

Reply:
I like that idea!